src/pinned_memory_manager.cc

```diff
@@ -153,6 +155,15 @@ PinnedMemoryManager::AllocInternal(
     if (status.IsOk()) {
       auto res = memory_info_.emplace(
           *ptr, std::make_pair(is_pinned, pinned_memory_buffer));
+
+      {
+        std::lock_guard<std::mutex> lk(alloc_info_mtx_);
```

Comment:
Can you move this within the `if` block and remove the extra `{ }` scope brackets, if `is_pinned` doesn't need to be protected?